lib/soft_deletion/core.rb

```diff
@@ -92,6 +92,13 @@ def soft_delete_dependencies
       self.class.soft_delete_dependents.map { |dependent| SoftDeletion::Dependency.new(self, dependent) }
     end
 
+    def update_counter_cache_associations(increment=false)
```

Comment:
should not be a public method ...
should be prefixed with soft_deletion or somehow have it in it's name ... 
also would prefer this as a single method ... logic is not so complicated it needs it's own class ...